### PR TITLE
Simplify List() signature in clients.

### PR DESCRIPTION
**Pull request by wojtek-t**
This is a sibling PR to #17863 

@krousey @lavalamp

### CODE REVIEW

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

**Comment by lavalamp** [comment]
@caesarxuchao Can you review this one? Thanks :)

**Comment by wojtek-t** [comment]
@caesarxuchao - it would be awesome if you could do it. Thanks!

**Comment by caesarxuchao** [comment]
Sure, I'll review it later today.

File: pkg/kubectl/rolling_updater.go [hunk 1]
```diff
@@ -678,7 +680,9 @@ func AddDeploymentKeyToReplicationController(oldRc *api.ReplicationController, c
 	// Clean up any orphaned pods that don't have the new label, this can happen if the rc manager
 	// doesn't see the update to its pod template and creates a new pod with the old labels after
 	// we've finished re-adopting existing pods to the rc.
-	podList, err = client.Pods(namespace).List(labels.SelectorFromSet(selectorCopy), fields.Everything(), unversioned.ListOptions{})
+	selector = labels.SelectorFromSet(selectorCopy)
```
Comment (caesarxuchao) [inline pkg/kubectl/rolling_updater.go]: Out of curiosity, did you automate this kind of changes? If so, how did you do that?

**Comment by caesarxuchao** [comment]
LGTM. Thanks. It needs rebase.

File: pkg/kubectl/rolling_updater.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/kubectl/rolling_updater.go]: No - I did it manually.
[It was easy to fix it with sed when both selectors was Everything() - all other changes I did manually].

**Comment by wojtek-t** [comment]
Trivial rebase - reapplying lgtm

**Comment by wojtek-t** [comment]
One more trivial rebase

**Comment by gmarek** [comment]
@k8s-bot unit test this

**Comment by wojtek-t** [comment]
unit-test failure is unrelated to this PR:
F1203 08:24:45.055467   17949 proxy.go:132] listen tcp 127.0.0.1:51245: bind: address already in use

@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
One more trivial rebase

**Pull request merged**